src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/CursorIT.java

```diff
@@ -0,0 +1,317 @@
+        for(int i =0 ; i < 5 ; i++) {
+            closeResp = executeCursorCloseQuery(cursor);
+            assertThat(closeResp.getBoolean("succeeded"), equalTo(true));
+        }
+
+        // using the cursor after its cleared, will throw exception
+        Response response = null;
+        try {
+            JSONObject queryResult = executeCursorQuery(cursor);
+        } catch (ResponseException ex) {
+            response = ex.getResponse();
+        }
+
+        JSONObject resp = new JSONObject(TestUtils.getResponseBody(response));
+        assertThat(resp.getInt("status"), equalTo(404));
+        assertThat(resp.query("/error/reason"), equalTo("all shards failed"));
+        assertThat(resp.query("/error/caused_by/reason").toString(), containsString("No search context found"));
+        assertThat(resp.query("/error/type"), equalTo("search_phase_execution_exception"));
+    }
+
+
+    @Test
+    public void invalidCursorIdNotDecodable() throws IOException {
+        // could be either not decode-able
+        String randomCursor = "eyJzY2hlbWEiOlt7Im5hbWUiOiJmaXJzdG5hbWUiLCJ0eXBlIjoidGV4dCJ9LHsibmFtZSI6InN0Y";
+
+        Response response = null;
+        try {
+            JSONObject resp = executeCursorQuery(randomCursor);
+        } catch (ResponseException ex) {
+            response = ex.getResponse();
+        }
+
+        JSONObject resp = new JSONObject(TestUtils.getResponseBody(response));
+        assertThat(resp.getInt("status"), equalTo(400));
+        assertThat(resp.query("/error/type"), equalTo("illegal_argument_exception"));
+    }
+
+    @Test
+    public void respectLimitPassedInSelectClause() throws IOException {
+        //TODO:
+//        String query = StringUtils.format("SELECT firstname, email, state FROM %s LIMIT 800", TEST_INDEX_ACCOUNT);
+//        String result = executeFetchQuery(query, 50, "jdbc");
+        assertThat(1, equalTo(1));
+    }
+
+
+    @Test
+    public void noPaginationWithNonJDBCFormat() throws IOException {
+        // checking for CSV, RAW format
+        String query = StringUtils.format("SELECT firstname, email, state FROM %s LIMIT 2000", TEST_INDEX_ACCOUNT);
+        String csvResult = executeFetchQuery(query, 100, "csv");
+        String[] rows = csvResult.split("\n");
+        // all the 1000 records (+1 for header) are retrieved instead of fetch_size number of records
+        assertThat(rows.length, equalTo(1001));
+
+        String rawResult = executeFetchQuery(query, 100, "raw");
+        rows = rawResult.split("\n");
+        // all the 1000 records (NO headers) are retrieved instead of fetch_size number of records
+        assertThat(rows.length, equalTo(1000));
+    }
+
+
+    public void verifyWithAndWithoutPaginationResponse(String sqlQuery, String cursorQuery, int fetch_size) throws IOException {
+        // we are only checking here for schema and aatarows
+        JSONObject withoutCursorResponse = new JSONObject(executeFetchQuery(sqlQuery, 0, JDBC));
```

Comment:
Is fetch_size the mandatory field for query?, if not, we don't need include it.

Reply:
No, its not mandatory. There is another version `executeFetchLessQuery` without passing the `fetch_size`.  We passed `0` as `fetch_size` here for default non-paginated behavior.